hypertrace-core-graphql-log-event-schema/src/main/java/org/hypertrace/core/graphql/log/event/dao/GatewayServiceLogEventsResponseConverter.java

```diff
@@ -0,0 +1,64 @@
+package org.hypertrace.core.graphql.log.event.dao;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.common.request.AttributeRequest;
+import org.hypertrace.core.graphql.common.utils.BiConverter;
+import org.hypertrace.core.graphql.log.event.request.LogEventRequest;
+import org.hypertrace.core.graphql.log.event.schema.LogEvent;
+import org.hypertrace.core.graphql.log.event.schema.LogEventResultSet;
+import org.hypertrace.gateway.service.v1.common.Value;
+import org.hypertrace.gateway.service.v1.log.events.LogEventsResponse;
+
+public class GatewayServiceLogEventsResponseConverter {
+
+  private final BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+      attributeMapConverter;
+
+  @Inject
+  GatewayServiceLogEventsResponseConverter(
+      BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+          attributeMapConverter) {
+    this.attributeMapConverter = attributeMapConverter;
+  }
+
+  public Single<LogEventResultSet> convert(LogEventRequest request, LogEventsResponse response) {
+    return Observable.fromIterable(response.getLogEventsList())
+        .flatMapSingle(logEvent -> this.convert(request, logEvent))
+        .toList()
+        .map(
+            logEvents ->
+                new ConvertedLogEventResultSet(logEvents, logEvents.size(), logEvents.size()));
```

Comment:
could you leave a TODO for total

Reply:
Do we want to populate this `total` field or have a separate api like `getTotalLogEvent` to fetch `total`?

Reply:
GQL doesn't need a separate api since it can selectively populate this based on whether the call requested the total.